src/masternode-payments.cpp

```diff
@@ -260,14 +260,16 @@ int CMasternodePayments::GetMinMasternodePaymentsProto() {
 
 void CMasternodePayments::ProcessMessage(CNode* pfrom, std::string& strCommand, CDataStream& vRecv)
 {
-    if(!masternodeSync.IsBlockchainSynced()) return;
-
-    if(fLiteMode) return; //disable all Darksend/Masternode related functionality
+    // Ignore any payments messages until masternode list is synced
+    if(!masternodeSync.IsMasternodeListSynced()) return;
```

Comment:
Same here, avoiding useless "unknown masternode" and AskForMN() calls. Here it's also a replacer for removed 297-299, it just fails much earlier without reading any message or performing any additional (and in this case useless) checks.